src/platform/FailSafeContext.cpp

```diff
@@ -22,17 +22,31 @@
 
 #include <platform/FailSafeContext.h>
 
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <lib/support/SafeInt.h>
 #include <platform/ConfigurationManager.h>
 
 namespace chip {
 namespace DeviceLayer {
 
+namespace {
+constexpr TLV::Tag kFabricIndexTag      = TLV::ContextTag(0);
+constexpr TLV::Tag kAddNocCommandTag    = TLV::ContextTag(1);
+constexpr TLV::Tag kUpdateNocCommandTag = TLV::ContextTag(2);
+} // anonymous namespace
+
 void FailSafeContext::HandleArmFailSafe(System::Layer * layer, void * aAppState)
 {
     FailSafeContext * context = reinterpret_cast<FailSafeContext *>(aAppState);
     context->FailSafeTimerExpired();
 }
 
+void FailSafeContext::HandleDisarmFailSafe(intptr_t arg)
+{
+    ConfigurationMgr().SetFailSafeArmed(false);
```

Comment:
But this can race with a new failsafe being armed, since we set `mFailSafeArmed` to false already, so a new one can be armed before this async event is processed. It seems to me that all the disarming work (including setting mFailSafeArmed to false) needs to happen at the same time.

Reply:
That will go back to my previous version, if all disarming works happen at the same time, but cleanup event is processed asynchronously, what happens if device crash or shut down before the event is processed? That will skip the cleanup, right?